decision-records/010-remove-init-method.md

```diff
@@ -0,0 +1,98 @@
+# Remove `init()` method of JavaScript components
+<!-- The title should be a short present tense imperative phrase, less than 50
+     characters, like a git commit message. -->
+
+**Date:** 2023-07-20
+
+**Status:** Accepted
+
+## Decision
+
+<!-- An overview of the decision made. -->
+
+We'll be removing the `init()` method of our JavaScript components,
+in favour of them initialising automatically during construction.
+
+This simplifies individual component's initialisation from:
+
+```js
+new Accordion($element).init()
```

Comment:
I think it'd be worth migrating away from `$module`to avoid confusions as we're dealing with proper modules (ES Modules) going forward, hence the `$element`.

Reply:
Sounds like a GitHub issue 😉

Reply:
Definitely. We may have one already, though, as the topic sounds like something we already mentioned, I think when sorting the scope of v5, but set to work on later.